modules/service/src/main/java11/org/mule/runtime/module/service/api/artifact/ServiceModuleLayerFactory.java

```diff
@@ -97,11 +98,19 @@ public ArtifactClassLoader create(String artifactId, ServiceDescriptor descripto
       propagateOpensToService(parent, artifactLayer, serviceModuleAnnotationClass, serviceModule, serviceModuleName);
     }
 
-    return new MuleServiceClassLoader(artifactId,
-                                      descriptor,
-                                      new URL[0],
-                                      artifactLayer.findLoader(serviceModuleName),
-                                      lookupPolicy);
+    ArtifactClassLoader serviceClassLoader =
+        new MuleServiceClassLoader(artifactId, descriptor, new URL[0], artifactLayer.findLoader(serviceModuleName), lookupPolicy);
+
+    ModuleLayerGraph.setModuleLayerId(artifactLayer, artifactId);
+    artifactLayer.parents().stream().filter(parentLayer -> !parentLayer.equals(ModuleLayer.boot())).findFirst()
```

Comment:
use `anyMatch` instead of `findFirst/ifPresent`

Reply:
`anyMatch` returns `boolean`, and I need the object to assign the name